consensus/src/state_computer.rs

```diff
@@ -36,6 +34,18 @@ type NotificationType = (
     Vec<ContractEvent>,
 );
 
+#[derive(Clone, Copy, Debug, Eq, PartialEq, PartialOrd, Ord, Hash)]
+struct LogicalTime {
```

Comment:
not need anymore?

Reply:
It's still used below for avoiding doing the sync_to again. But it's only used in this file.